content/en/docs/concepts/extend-kubernetes/api-extension/custom-resources.md

```diff
@@ -161,7 +161,7 @@ The [CustomResourceDefinition](/docs/tasks/extend-kubernetes/custom-resources/cu
 API resource allows you to define custom resources.
 Defining a CRD object creates a new custom resource with a name and schema that you specify.
 The Kubernetes API serves and handles the storage of your custom resource.
-The name of a CRD object must be a valid
+The name of an object whose kind/resource is defined by a CRD must be a valid
```

Comment:
Sounds to me that this paragraph focuses more on the CRD rather than the CRs created from the CRDs.

Reply:
BTW, regarding the persistent slippage between "resource" and "object", this file has a clear statement that comports with _most_ (but sadly not all) usage in the code:

> A resource is an endpoint in the [Kubernetes API](https://github.com/kubernetes/website/blob/b28ffc2d45545e5acfd6c401bb5a5ba15c6a041a/docs/concepts/overview/kubernetes-api) that stores a collection of {{< glossary_tooltip text="API objects" term_id="object" >}} of a certain kind; for example, the built-in pods resource contains a collection of Pod objects.

Reply:
> A resource is an endpoint in the [Kubernetes API](https://github.com/kubernetes/website/blob/b28ffc2d45545e5acfd6c401bb5a5ba15c6a041a/docs/concepts/overview/kubernetes-api) that stores a collection of {{< glossary_tooltip text="API objects" term_id="object" >}} of a certain kind; for example, the built-in pods resource contains a collection of Pod objects.

That text isn't as helpful as we'd like it to be, because (at the HTTP level) all objects are also resources. In fact, and as I see it:
- all collections are resources
- all objects are resources
- some resources are objects

You can use a CRD to define resources that aren't objects, so we should use “resource” when we talk about those. When we talk about CRDs themselves, we can call them resources but every CRD itself is 100% an object.

Reply:
Yes, I agree, in the web world in general, "resource" is quite a general thing. However, in most of the Kubernetes code, "resource" is specifically a collection of objects. In any case, my real problem here is understanding where I should put the remark about names of objects whose kind or resource is defined by a CRD.

Reply:
> where I should put the remark about names of objects whose kind or resource is defined by a CRD.

The page could use a refactor for sure. However, given how it looks today, I'd put that remark within
https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions

We could use a tutorial page about writing (and installing) a CRD; something for another time though.

Reply:
Also, I am not sure I understand the remark that "You can use a CRD to define resources that aren't objects". Terminology still gets in the way of even that. If we start with the dichotomy that I quoted 30 minutes ago (between "resource" and "object"), then it is a tautology that a CRD defines a resource and a resource is not an object. But I suspect you mean that a CRD can define a collection of things that are not objects? If so, that is a surprisse to me; can you please elaborate?

Reply:
Ignore what's in the comments and source code for a while; those aren't really part of the documentation.

A CRD can define a new API for [things with `.metadata` and `.spec`], which for short we call “objects”. So you could add a CRD and get a CompressedConfigMap API, and then create three CompressedConfigMaps.

I think you could also use a CRD to define an API for [things that have `.fruitName`] and no other fields. Those things would not be K8s objects, but they would be K8s resources. Sets of the things, outside of the Golang implementation of Kubernetes' API, would be collections (for example, **deletecollection** works on them).

The CompressedConfigMaps would be resources and the fruit things would be resources too, but the fruit things wouldn't be objects.